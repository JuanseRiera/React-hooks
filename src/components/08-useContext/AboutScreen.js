import React, { useContext } from "react";
import { userContext } from "./UserContex";

export const AboutScreen = () => {
  const { value, setvalue } = useContext(userContext);
  return (
    <>
      <h1>AboutScreen</h1>
      <hr />

      <pre>{JSON.stringify(value, null, 3)}</pre>

      <button
        className="btn btn-danger"
        onClick={() => {
          setvalue({});
        }}
      >
        Logout
      </button>
    </>
  );
};
