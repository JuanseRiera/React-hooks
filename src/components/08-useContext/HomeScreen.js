import React, { useContext } from "react";
import { userContext } from "./UserContex";

export const HomeScreen = () => {
  const { value } = useContext(userContext);
  console.log(value);
  return (
    <>
      <h1>HomeScreen</h1>
      <hr />

      <pre>{JSON.stringify(value, null, 3)}</pre>
    </>
  );
};
