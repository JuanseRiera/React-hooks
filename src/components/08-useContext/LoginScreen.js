import React, { useContext } from "react";
import { userContext } from "./UserContex";

export const LoginScreen = () => {
  const { setvalue } = useContext(userContext);
  console.log(setvalue);
  return (
    <>
      <h1>LoginScreen</h1>
      <hr />

      <button
        className="btn btn-info"
        onClick={() => {
          setvalue({
            id: 123,
            nombre: "Juanse",
          });
        }}
      >
        Login
      </button>
    </>
  );
};
