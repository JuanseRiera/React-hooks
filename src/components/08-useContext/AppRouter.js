import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
  return (
    <>
      <Router>
        <div>
          <NavBar />
          <Switch>
            <Route path="/login" component={LoginScreen} />
            <Route path="/about" component={AboutScreen} />
            <Route path="/" component={HomeScreen} />
          </Switch>
        </div>
      </Router>
    </>
  );
};
