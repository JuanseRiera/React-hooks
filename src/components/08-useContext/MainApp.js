import React, { useState } from "react";
import { AppRouter } from "./AppRouter";
import { userContext } from "./UserContex";

export const MainApp = () => {
  const [value, setvalue] = useState({});
  return (
    <>
      <userContext.Provider value={{ value, setvalue }}>
        <AppRouter />
      </userContext.Provider>
    </>
  );
};
