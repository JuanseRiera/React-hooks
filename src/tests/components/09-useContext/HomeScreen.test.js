import { mount } from "enzyme";
import { HomeScreen } from "../../../components/08-useContext/HomeScreen";
import { userContext } from "../../../components/08-useContext/UserContex";
describe("Pruebas del homescreen", () => {
  const user = {
    name: "juanse",
    email: "rierajuanse@gmail.com",
  };
  const wrapper = mount(
    <userContext.Provider value={{ user }}>
      <HomeScreen />
    </userContext.Provider>
  );
  test("Deberia de cargar correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
