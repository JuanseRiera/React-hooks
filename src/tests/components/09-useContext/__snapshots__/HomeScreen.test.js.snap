// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del homescreen Deberia de cargar correctamente 1`] = `
Array [
  <h1>
    HomeScreen
  </h1>,
  <hr />,
  <pre />,
]
`;
